### PR TITLE
#622 Migrate Courses - Course show page - Copy content from other course

### PR DESCRIPTION
**Pull request by goodviber**
### Context
The "Courses" section (linked from a page like this https://qa.publish-teacher-training-courses.service.gov.uk/organisations/2GU/2022) needs to be reimplemented in New Publish.

What needs to be done
Reimplement the functionality to allow providers to copy content from another course when editing it.
Tech Notes
Remember to add the partials to enable this for every sub section of the courses show page which allow it

course_list
### Guidance to review
Should have tests for copy.rb, and fetch.rb
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by goodviber** [comment]
[Migrate Courses - Course show page - Copy content from another course](https://trello.com/c/x9R7klsz/622-migrate-courses-course-show-page-copy-content-from-another-course)

File: app/views/publish/courses/course_information/edit.html.erb [hunk 1]
```diff
@@ -115,11 +121,21 @@
...
-        label: { text: @course.placements_heading, size: "s" },
+        value: @course.enrichments.last[:how_school_placements_work],
+        label: { text: "How school placements work", size: "s" },
         max_words: 350,
         rows: 15) %>
 
       <%= f.govuk_submit "Save" %>
     <% end %>
   </div>
+  <aside class="govuk-grid-column-one-third" data-qa="course__related_sidebar">
```
Comment (darokel) [inline app/views/publish/courses/course_information/edit.html.erb:132]: I think we need to add this to nearly every sub section in the courses show page. Eg every one of these sub sections with change links allow you to copy info from another course. Might be worth double checking and adding the partial in the remaining places? 
 
Screenshot 2022-03-28 at 09 58 42

File: app/views/publish/courses/_course_length_field.html.erb [hunk 2]
```diff
@@ -4,22 +4,17 @@
     id: form_object.errors.key?(:course_length) ? "course_length-error" : "course-length",
   }) do %>
     <%= f.govuk_radio_button(:course_length, "OneYear",
+      checked: @course.enrichments.last[:course_length] == "OneYear",
```
Comment (goodviber) [inline app/views/publish/courses/_course_length_field.html.erb:7]: @course.course_length ?

File: spec/features/publish/courses/gcse_equivalency_spec.rb [hunk 3]
```diff
@@ -15,8 +15,10 @@
 
     then_i_fill_the_equivalency_requirements
     and_i_click_save
-    and_i_am_on_the_course_page
-    and_i_see_the_success_summary
+    # this is not working with the form changes
+    # binding.pry
```
Comment (goodviber) [inline spec/features/publish/courses/gcse_equivalency_spec.rb:19]: Am still working on this...

Comment (goodviber) [reply spec/features/publish/courses/gcse_equivalency_spec.rb]: Done

File: app/views/publish/courses/course_information/edit.html.erb [hunk 4]
```diff
@@ -73,13 +78,14 @@
...
       <%= f.govuk_text_area(:interview_process,
+        value: @course.enrichments.last[:interview_process],
         label: { text: "Interview process (optional)", size: "s" },
         max_words: 250,
         rows: 15) %>
 
       <hr class="govuk-section-break govuk-section-break--visible govuk-section-break--l">
 
-      <h3 class="govuk-heading-m" id="how-school-placements-work"><%= @course.placements_heading %></h3>
+      <h3 class="govuk-heading-m" id="how-school-placements-work">How school placements work</h3>
```
Comment (defong) [inline app/views/publish/courses/course_information/edit.html.erb:88]: What's the rationale?
`@course.placements_heading` That is used a fair bit.
Do we need to remove it wholesomely?

File: app/controllers/publish/courses/course_information_controller.rb [hunk 5]
```diff
@@ -1,10 +1,15 @@
 module Publish
   module Courses
     class CourseInformationController < PublishController
+      include CourseBasicDetailConcern
```
[the remark this answers is not in the document]
Comment (goodviber) [reply app/controllers/publish/courses/course_information_controller.rb]: Yep

File: app/views/publish/courses/course_information/edit.html.erb (hunk 4 above)
Comment (goodviber) [reply app/views/publish/courses/course_information/edit.html.erb]: No, was an oversight on my part, replaced

File: app/views/publish/courses/_course_length_field.html.erb (hunk 2 above)
Comment (goodviber) [reply app/views/publish/courses/_course_length_field.html.erb]: @course.course_length works here, but not in courses/course_information/edit.html.erb

Comment (goodviber) [reply app/views/publish/courses/_course_length_field.html.erb]: Actually it has started working, I hate these tests...

**Review by defong — APPROVED** [review]

**Pull request merged**